homeassistant/components/simplisafe/__init__.py

```diff
@@ -200,3 +184,33 @@ def _async_save_refresh_token(hass, config_entry, token):
     remove_listener()
 
     return True
+
+
+class SimpliSafe:
+    """Define a SimpliSafe API object."""
+
+    def __init__(self, hass, config_entry, systems):
+        """Initialize."""
+        self._config_entry = config_entry
+        self._hass = hass
+        self.systems = systems
+
+    async def async_update(self):
+        """Get updated data from SimpliSafe."""
+        tasks = [system.update() for system in self.systems.values()]
+
+        results = await asyncio.gather(*tasks, return_exceptions=True)
+        for system, result in zip(self.systems.values(), results):
+            if isinstance(result, SimplipyError):
```

Comment:
I would check for type `Exception` given the logic is not dependent on the type of exception.  Even though the library may only raise `SimplipyError`, you could get other exceptions if the task was canceled.